src/basic/string-util.c

```diff
@@ -1509,24 +1509,20 @@ ssize_t strlevenshtein(const char *x, const char *y) {
 }
 
 char *strrstr(const char *haystack, const char *needle) {
-        const char *f = NULL;
-        size_t l;
-
         /* Like strstr() but returns the last rather than the first occurrence of "needle" in "haystack". */
 
         if (!haystack || !needle)
                 return NULL;
 
-        l = strlen(needle);
-
         /* Special case: for the empty string we return the very last possible occurrence, i.e. *after* the
          * last char, not before. */
-        if (l == 0)
-                return strchr(haystack, 0);
+        if (*needle == 0)
+                return (char *) haystack + strlen(haystack);
```

Comment:
why change this?

Reply:
To avoid having to call strlen

Reply:
Please drop the change for the `return` line. You do not answer the question, and in the changed line you call `strlen()`...